# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/generic/LocalAccessible.h
# Commit: fca1ff7eabf4
# Full Hash: fca1ff7eabf4ddfca60184253136535d2285488d
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2022-01-05 09:33:53
# Regressor Bug: 1737944
# File Overlap Count: 2
# Description:
#   Bug 1737944 - P6: Put AccGroupInfo getter/creator in Accessible. r=Jamie
#   
#   Changed the methods a bit, instead of a HasDirty boolean method, I went
#   with a getter and a get-and-creator. This will lend better to remote
#   accessibles where we won't have a dirty flag.
# ==============================================================================

diff -r c0a7fdc4bfca -r fca1ff7eabf4 accessible/generic/LocalAccessible.h
--- a/accessible/generic/LocalAccessible.h	Tue Jan 04 21:01:36 2022 +0000
+++ b/accessible/generic/LocalAccessible.h	Tue Jan 04 21:01:36 2022 +0000
@@ -732,13 +732,6 @@
   }
 
   /**
-   * Return true if the accessible's group info needs to be updated.
-   */
-  inline bool HasDirtyGroupInfo() const {
-    return mStateFlags & eGroupInfoDirty;
-  }
-
-  /**
    * Return true if the accessible has associated DOM content.
    */
   bool HasOwnContent() const {
@@ -1024,10 +1017,9 @@
    */
   uint32_t GetActionRule() const;
 
-  /**
-   * Return group info.
-   */
-  AccGroupInfo* GetGroupInfo() const;
+  virtual AccGroupInfo* GetGroupInfo() const override;
+
+  virtual AccGroupInfo* GetOrCreateGroupInfo() override;
 
   virtual void ARIAGroupPosition(int32_t* aLevel, int32_t* aSetSize,
                                  int32_t* aPosInSet) const override;
